### DISCUSSION TITLE
rejected

### DISCUSSION THREAD

**Discussion by Zeekdrich** [community/community#34594] [Pull Requests]
Please how do i resolve such error  ! [rejected]        master -> master (non-fast-forward)
error: failed to push some refs
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

**Comment by airtower-luna** [comment]
That means the remote `master` branch contains commits that *aren't* in the local branch you're trying to push, and would get lost if the push went ahead.

What to do depends on whether you actually want to keep those commits:

* If you want to keep them, either merge the remote branch into your local one, or rebase your local branch on the remote one. Then push.
* If you actually want to discard those commits you can push with `--force` to tell Git it should push even if it might mean losing data. :warning:

Comparing the branches with `git diff` and `git log` might help you decide. :wink:

**Reply by Zeekdrich** [reply]
This is helpful. Thanks a lot

**Reply by pythonpioneer** [reply]
Mark it as answered.
